clients/fides-js/src/fides.ts

```diff
@@ -0,0 +1,86 @@
+/***
+ * TODO: description
+ * TODO: preferably reconfigure tsup to generate fides.js instead of fides.global.js
+ * 
+ * Expected usage of this in an HTML file is:
+ * ```
+ * <script src="../dist/fides.global.js"></script>
+ * <script>
+ *   window.Fides.init({
+ *     consent: {
+ *       options: [...(default consent options here)]
+ *     }
+ *   });
+ * </script>
+ * ```
+ */
+import { gtm } from "./integrations/gtm";
+import { meta } from "./integrations/meta";
+import { shopify } from "./integrations/shopify";
+import { ConsentConfig } from "./lib/consent-config";
+import { getConsentContext } from "./lib/consent-context";
+import { CookieKeyConsent, getConsentCookie, makeDefaults } from "./lib/cookie";
+
+/**
+ * Configuration options for the fides.js script
+ */
+export interface FidesConfig {
+  consent: ConsentConfig;
+}
+
+type Fides = {
+  consent: CookieKeyConsent;
+  gtm: typeof gtm;
+  init: typeof init;
+  initialized: boolean;
+  meta: typeof meta;
+  shopify: typeof shopify;
+}
+
+declare global {
+  interface Window {
+    Fides: Fides;
+  }
+}
+
+/**
+ * Initialize the Fides object with configuration values
+ */
+const init = (config: FidesConfig) => {
```

Comment:
This is changed from `fides-consent` where all this initialization logic ran immediately. It was able to do that because it could import the "consent-config.json" object at build-time, but we have to cut that dependency out.

Therefore, we now need to defer the initialization of the consent defaults until runtime when the config options are provided. That's what this `init` function is for 👍

Reply:
this makes sense